website/source/docs/acl/acl-identity-providers.html.md

```diff
@@ -0,0 +1,163 @@
+
+# ACL Identity Providers
+
+An ACL Identity Provider is a trusted external party that can be configured to
+to allow applications to authenticate to Consul using existing credentials and
+receive ACL Tokens usable within the local datacenter in exchange.
+
+The only supported type of identity provider in Consul 1.5 is `kubernetes`
+but it is expected that more will come later.
+
+## Overview
+
+Without an identity provider integration, a trusted operator needs to be
+critically involved in the creation and secure introduction of each ACL Token
+to every application that needs one, and ensure that the policies assigned to
+these tokens follow the principle of least-privilege.
+
+When running in environments such as a public cloud or when supervised by a
+cluster scheduler, applications may already have access to uniquely identifying
+credentials that were delivered securely by the platform. Consul identity
+provider integrations allow for these credentials to be used to generate ACL
+Tokens with properly-scoped policies without operator intervention.
+
+In Consul 1.5 the focus is around simplifying the creation of tokens with the
+privileges necessary to participate in a [Connect](/docs/connect/index.html)
+service mesh with minimal operator intervention.
+
+## Operator Configuration
+
+An operator needs to configure each identity provider that is to be trusted by
+using the API or command line before they can be used by applications.
+
+* **Authentication** - Details about how to authenticate application
+  credentials are configured using the `consul acl idp` subcommands or the
+  corresponding [API endpoints](/api/acl/identity-providers.html). The specific
+  details of configuration are type dependent and described below.
+
+* **Authorization** - One or more Role Binding Rules must be configured
+  definiting how to translate trusted identity attributes into privileges
+  assigned to the ACL Token that is created. These can be managed with the
+  `consul acl rolebindingrule` subcommands or the corresponding [API
+  endpoints](/api/acl/role-binding-rules.html).
+
+## Login Process
+
+1. Applications can use the `consul acl login` subcommand or the [login API
+   endpoint](/api/acl/acl.html#login-to-identity-provider) to authenticate to
+   an identity provider through the Consul leader.
+
+2. The identity provider validates the credentials and returns trusted identity
+   attributes to the Consul leader.
+
+3. The Consul leader consults the configured set of Role Binding Rules linked
+   to the identity provider to find rules that match the trusted identity
+   attributes.
+
+4. If any Role Binding Rules match an ACL Token is created in the local
+   datacenter and linked to the computed Roles.
+
+5. Applications can use the `consul acl logout` subcommand or the [logout API
+   endpoint](/api/acl/acl.html#logout-from-identity-provider) to destroy their
+   token when it is no longer required.
+
+## Kubernetes Identity Provider
```

Comment:
I'm feeling this section should probably be on it's own page probably still under ACLs category. Originally, I thought it should live with the other Kubernetes docs but then where would Vault go?